bowlingGame.js

```diff
@@ -0,0 +1,65 @@
+class BowlingGame {
+  constructor(){
+    this.rolls = [];
+    this.currentFrame = 1;
+  };
+  roll(number){
+    if(number > 10){
+      throw "Invalid Roll";
+    }
+    else {
+      this.rolls.push( { value: number, frame: this.currentFrame } );
+    };
+    
+  };
+  // [10, 5, 5, 10, 5, 3]
+  score(){
+    let totalScore = 0;
+    let currentRoll = 0;
+    while(currentRoll < this.rolls.length){
+      if(this.currentFrame == 10){
+        totalScore += this.sumTheRest(currentRoll);
+        break;
+      }
+      else if(this.frameIsStrike(currentRoll)){       
+        totalScore += this.bonusScoring(currentRoll);
+        currentRoll += 1;
+        this.currentFrame += 1
+      }
+      else if(this.frameIsSpare(currentRoll)){
+        totalScore += this.bonusScoring(currentRoll);
```

Comment:
There is one bonusScoring method for both the strike and spare scenarios. It looks like the spare bonus is then incorrect because you're adding the values of the next two rolls in the bonusScoring method. The spare bonus should only be the first roll in the next frame.